development/tools/cmd/pjtester/README.md

```diff
@@ -2,76 +2,82 @@
 
-Finally, `pjtester` creates the Prow Job on the production Prow instance. The Prow Job name for which you triggered the test is prefixed with `{YOUR_GITHUB_USER}_test_of_prowjob_`.
+Once prowjob definition is found and loaded, `pjtester` generates prowjob specification. Prowjob name and context reported to GitHub are prefixed with the pjtester prefix. Prowjob refs and extraRefs are set according to the configuration provided in the `pjtester.yaml` file in pjtester prowjob.
 
-Because the file `vpath/pjtester.yaml` is used by `pjtester` only to know the name of the Prow Job to test, it should not exist outside of the PR. This is why the `pre-main-test-infra-vpathgurad` required context was added. Its simple task is to fail whenever the `vpath` directory exists and to prevent the PR merge. As soon as the virtual path disappears from the PR, `vpathguard` will allow for the PR merge.
+If the `pjtester.yaml` file contains pull request numbers in the **prConfigs** parameter, they are used as prowjob refs and extraRefs.
 
-## Prerequisites
+If **prConfigs** doesn't provide a pull request number for refs or some extraRefs, but pjtester pull request is open on the same repository, it is used in the prowjob specification as refs or extraRefs.
 
-Prow Job tester is a tool running on Prow. You need a working Prow instance to use it.
+If some extraRefs are not set in the previous steps, they will be set to values loaded from the source. If prowjob refs are not set, `pjtester` will set it to match the repository `heads/main` details for postsubmit. Presubmit refs are set to match the latest pull request merged to `main` branch.
 
-## Installation
+For presubmit jobs, Prow requires the pull request's head SHA, pull request number, and pull request author set in the Prow Job refs. In the `pjtester.yaml` file, you can specify a pull request number for a repository against which a tested Prow Job is running. If you don't specify it, `pjtester` finds the latest pull request merged to the`main` branch and uses its details for the presubmit refs.
 
-To make `pjtester` work for you, you need to compile it and build an image with its binary included. This is done by the `post-test-infra-prow-tools` Prow Job. It builds and pushes the `prow-tools` image.
-Next, you must add a presubmit job to trigger `pjtester` execution. This is done by the `pre-main-test-infra-pjtester` Prow Job.
+Finally, `pjtester` creates the ProwJob object on the production Prow instance Kubernetes cluster. The Prow Job name, for which you triggered the test, is prefixed with `{YOUR_GITHUB_USER}_test_of_prowjob_`.
 
-## Configuration
+Because the `vpath/pjtester.yaml` file is used by `pjtester` only, it should not exist outside the PR. This is why the `pre-vpathgurad` required context is added. It fails whenever the `vpath` directory exists and prevents the PR merge. As soon as the virtual path disappears from the PR, `vpathguard` will allow for the PR merge.
```

Comment:
```suggestion
Because the `vpath/pjtester.yaml` file is used by `pjtester` only, it must not exist outside the PR. This is why the `pre-vpathgurad` required context is added. It fails whenever the `vpath` directory exists and prevents the PR merge. As soon as the virtual path disappears from the PR, `vpathguard` will allow for the PR merge.
```